test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,344 @@
+            PushNotificationRegistrationOption::PushTrigger | PushNotificationRegistrationOption::ComActivator,
+            c_fakeComServerId); // same clsid as app manifest
+
+        appToken = PushNotificationManager::RegisterActivator(info);
+    }
+    catch (...)
+    {
+        auto registerActivatorException = hresult_error(to_hresult());
+        hr = registerActivatorException.code();
+    }
+    return hr == E_INVALIDARG;
+}
+
+bool runUnitTest(std::string unitTest)
+{
+    switch (switchMapping[unitTest])
+    {
+    case UnitTest::channelRequestUsingNullRemoteId:
+        return ChannelRequestUsingNullRemoteId();
+
+    case UnitTest::channelRequestUsingRemoteId:
+        return ChannelRequestUsingRemoteId();
+
+    case UnitTest::multipleChannelRequestUsingSameRemoteId:
+        return MultipleChannelRequestUsingSameRemoteId();
+
+    case UnitTest::multipleChannelRequestUsingMultipleRemoteId:
+        return MultipleChannelRequestUsingMultipleRemoteId();
+
+    case UnitTest::threeChannelRequestUsingSameRemoteId:
+        return ThreeChannelRequestUsingSameRemoteId();
+
+    case UnitTest::activatorTest:
+        return ActivatorTest();
+
+    case UnitTest::registerActivatorNullDetails:
+        return RegisterActivatorNullDetails();
+
+    case UnitTest::registerActivatorNullClsid:
+        return RegisterActivatorNullClsid();
+
+    case UnitTest::unregisterActivatorNullToken:
+        return UnregisterActivatorNullToken();
+
+    case UnitTest::unregisterActivatorNullBackgroundRegistration:
+        return UnregisterActivatorNullBackgroundRegistration();
+
+    case UnitTest::multipleRegisterActivatorTest:
+        return MultipleRegisterActivatorTest();
+
+    default:
+        return false;
+    }
+}
+
+// Cleanup function for exiting test app.
+void UnregisterClsid()
+{
+    if (appToken)
+    {
+        ::CoRevokeClassObject(static_cast<DWORD>(appToken.Cookie()));
+        PushNotificationManager::UnregisterActivator(appToken, PushNotificationRegistrationOption::PushTrigger);
+    }
+
+    if (fakeToken)
+    {
+        ::CoRevokeClassObject(static_cast<DWORD>(fakeToken.Cookie()));
+        PushNotificationManager::UnregisterActivator(fakeToken, PushNotificationRegistrationOption::PushTrigger);
+    }
+
+}
+
+int main()
+{
+    initUnitTestMapping();
+    bool output = false;
+    auto scope_exit = wil::scope_exit([&] {
+        UnregisterClsid();
+        if (output)
+        {
+            // Signal TAEF that protocol was activated and valid.
+            signalPhase(c_testProtocolScheme_Packaged);
+        }
+        else
+        {
+            // Signal TAEF that the test failed
+            signalPhase(c_testFailureEventName);
+        }
+    });
+
+    PushNotificationActivationInfo info(
+        PushNotificationRegistrationOption::PushTrigger | PushNotificationRegistrationOption::ComActivator,
+        winrt::guid("ccd2ae3f-764f-4ae3-be45-9804761b28b2")); // same clsid as app manifest
```

Comment:
Shouldn't we use `c_comServerId`? There are many copies of this same GUID scattered through the sample.